### PR TITLE
implementer new controllers

### PR DESCRIPTION
**Pull request by tarkaiev**

### CODE REVIEW

File: src/main/java/movie/tickets/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public AuthenticationServiceImpl(UserService userService,
     @Override
     public User login(String email, String password) throws AuthenticationException {
         logger.info("User with email " + email + " is trying to login");
-        Optional<User> userFromDb = userService.findByEmail(email);
+        Optional<User> userFromDb = Optional.ofNullable(userService.findByEmail(email));
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/tickets/security/AuthenticationServiceImpl.java]:
```suggestion
        Optional<User> userFromDb = Optional.ofNullable(userService.findByEmail(email));
```
userService.findByEmail(email) will ever return null?

File: src/main/java/movie/tickets/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    @PostMapping("/complete")
+    public void completeOrder(@RequestParam String email) {
+        User user = userService.findByEmail(email);
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
+        List<Ticket> tickets = shoppingCart.getTickets();
+        orderService.completeOrder(tickets, user);
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> orderHistory(@RequestParam String email) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/tickets/controller/OrderController.java]:
```suggestion
    public List<OrderResponseDto> getOrderHistory(@RequestParam String email) {
```

File: src/main/java/movie/tickets/model/dto/ShoppingCartResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package movie.tickets.model.dto;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private List<Long> ticketIds;
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/tickets/model/dto/ShoppingCartResponseDto.java]: let's use TicketResponseDto

File: src/main/java/movie/tickets/model/dto/CinemaHallRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package movie.tickets.model.dto;
+
+public class CinemaHallRequestDto {
+    private int capacity;
```
Comment (dmytro-hryhoruk) [inline src/main/java/movie/tickets/model/dto/CinemaHallRequestDto.java]:
```suggestion
    private Integer capacity;
```

File: src/main/java/movie/tickets/security/AuthenticationServiceImpl.java (hunk 1 above)
Comment (tarkaiev) [reply src/main/java/movie/tickets/security/AuthenticationServiceImpl.java]: nope

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**